go/test/endtoend/vreplication/resharding_workflows_v2_test.go

```diff
@@ -244,6 +246,73 @@ func TestBasicV2Workflows(t *testing.T) {
+while moving a table with an auto-increment from sharded to unsharded.
+*/
+func testVSchemaForSequenceAfterMoveTables(t *testing.T) {
+	// at this point the unsharded product and sharded customer keyspaces are created by previous tests
+
+	// use MoveTables to move customer2 from product to customer using
+	currentWorkflowType = wrangler.MoveTablesWorkflow
+	err := tstWorkflowExec(t, defaultCellName, "wf2", sourceKs, targetKs,
+		"customer2", workflowActionCreate, "", "", "")
+	require.NoError(t, err)
+	time.Sleep(1 * time.Second)
+	err = tstWorkflowExec(t, defaultCellName, "wf2", sourceKs, targetKs,
+		"", workflowActionSwitchTraffic, "", "", "")
+	require.NoError(t, err)
+	err = tstWorkflowExec(t, defaultCellName, "wf2", sourceKs, targetKs,
+		"", workflowActionComplete, "", "", "")
+	require.NoError(t, err)
+
+	// sanity check
+	output, err := vc.VtctlClient.ExecuteCommandWithOutput("GetVSchema", "product")
+	require.NoError(t, err)
+	assert.NotContains(t, output, "customer2\"", "customer2 still found in keyspace product")
+	validateCount(t, vtgateConn, "customer", "customer2", 3)
+
+	// check that customer2 has the sequence tag
+	output, err = vc.VtctlClient.ExecuteCommandWithOutput("GetVSchema", "customer")
+	require.NoError(t, err)
+	assert.Contains(t, output, "\"sequence\": \"customer_seq2\"", "customer2 sequence missing in keyspace customer")
+
+	// ensure sequence is available to vtgate
+	num := 5
+	for i := 0; i < num; i++ {
+		execVtgateQuery(t, vtgateConn, "customer", "insert into customer2(name) values('a')")
+	}
+	validateCount(t, vtgateConn, "customer", "customer2", 3+num)
+
+	// use MoveTables to move customer2 back to product. Note that now the table has an associated sequence
+	err = tstWorkflowExec(t, defaultCellName, "wf3", targetKs, sourceKs,
+		"customer2", workflowActionCreate, "", "", "")
+	require.NoError(t, err)
+	time.Sleep(1 * time.Second)
+	err = tstWorkflowExec(t, defaultCellName, "wf3", targetKs, sourceKs,
+		"", workflowActionSwitchTraffic, "", "", "")
+	require.NoError(t, err)
+	err = tstWorkflowExec(t, defaultCellName, "wf3", targetKs, sourceKs,
+		"", workflowActionComplete, "", "", "")
+	require.NoError(t, err)
+
+	// sanity check
+	output, err = vc.VtctlClient.ExecuteCommandWithOutput("GetVSchema", "product")
+	require.NoError(t, err)
+	assert.Contains(t, output, "customer2\"", "customer2 not found in keyspace product ")
+
+	// check that customer2 still has the sequence tag
+	output, err = vc.VtctlClient.ExecuteCommandWithOutput("GetVSchema", "product")
+	require.NoError(t, err)
+	assert.Contains(t, output, "\"sequence\": \"customer_seq2\"", "customer2 still found in keyspace product")
+
+	// ensure sequence is available to vtgate
+	for i := 0; i < num; i++ {
+		execVtgateQuery(t, vtgateConn, "product", "insert into customer2(name) values('a')")
+	}
+	validateCount(t, vtgateConn, "product", "customer2", 3+num+num)
```

Comment:
Is it worth checking the values to be sure they are actually incrementing integers as we expect? Not sure, but came to mind

Reply:
Added a check to ensure that the sequence values are being correctly used.